src/core/thread/topology.hpp

```diff
@@ -862,6 +862,56 @@ class Child : public Neighbor, public IndirectSender::ChildInfo, public DataPoll
 
 #endif // #if OPENTHREAD_CONFIG_CHILD_SUPERVISION_ENABLE
 
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+    /**
+     * This method returns MLR state of an Ip6 multicast address.
+     *
```

Comment:
Let's also add a comment to mention the reference should be one which we get from the iterator.
```suggestion
     * 
     * @note The @p aAdddress reference MUST be from `IterateIp6Addresses()` or `AddressIterator`.
```